### PR TITLE
Add dogfood GH action

### PR DESCRIPTION
**Pull request by Tim-Pohlmann**
Code taken from
[Java](https://github.com/SonarSource/sonar-java/blob/master/.github/workflows/dogfood.yml)
[Python](https://github.com/SonarSource/sonar-python/blob/9a7d8fdfed3cca92a413f181e86c707eb67c7173/.github/workflows/dogfood.yml)

### CODE REVIEW

**Review by andrei-epure-sonarsource — COMMENTED** [review]
Could you have a look also at https://github.com/SonarSource/sonar-cpp/blob/master/.github/workflows/dogfood.yml ?

See comment from @Godin https://sonarsource.slack.com/archives/CFSLVKNBB/p1702033384401889?thread_ts=1702029512.276729&cid=CFSLVKNBB

File: .github/workflows/dogfood.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          SLACK_WEBHOOK: ${{ fromJSON(steps.secrets.outputs.vault).SLACK_WEBHOOK }}
+          SLACK_OVERRIDE_MESSAGE: 'Dogfood build for `${{ steps.dogfood.outputs.sha1 }}`: *successful*'
+        with:
+          args: 'Succeed to build dogfood branch'
+      - name: Notify failures on Slack
+        uses: Ilshidur/action-slack@1.6.2
+        if: failure()
+        env:
+          SLACK_WEBHOOK: ${{ fromJSON(steps.secrets.outputs.vault).SLACK_WEBHOOK }}
+          SLACK_OVERRIDE_MESSAGE: 'Dogfood build for `${{ steps.dogfood.outputs.sha1 }}`: *failed*, see the logs at https://github.com/SonarSource/sonar-iac/actions'
```
Comment (antonioaversa) [inline .github/workflows/dogfood.yml:50]: `https://github.com/SonarSource/sonar-iac/actions` seems a copy-paste issue.

File: .github/workflows/dogfood.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        run: echo "The dogfood branch was `${{ steps.dogfood.outputs.dogfood-branch }}` and its HEAD SHA1 was `${{ steps.dogfood.outputs.sha1 }}`"
+      #slack notifications
+      - name: Notify success on Slack
+        uses: Ilshidur/action-slack@1.6.2
+        env:
+          SLACK_WEBHOOK: ${{ fromJSON(steps.secrets.outputs.vault).SLACK_WEBHOOK }}
+          SLACK_OVERRIDE_MESSAGE: 'Dogfood build for `${{ steps.dogfood.outputs.sha1 }}`: *successful*'
+        with:
+          args: 'Succeed to build dogfood branch'
+      - name: Notify failures on Slack
```
Comment (antonioaversa) [inline .github/workflows/dogfood.yml:45]: Nitpick: `success` on line 38 is singular, but `failures` here is plural.
This seems to reflect the saying that misfortune never comes alone :-)
Anyway, you can ignore this comment and keep it as is, if you want to conform to the yml of the other repos, that have the same message.

File: .github/workflows/dogfood.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  SLACK_CHANNEL: team-lang-dotnet
+  SLACK_USERNAME: Dogfood build action
+
+jobs:
+  dogfood_merge:
+    runs-on: ubuntu-latest
+    name: Update dogfood branch
+    permissions:
+      id-token: write # required for SonarSource/vault-action-wrapper
+      contents: write # required to grant GITHUB_TOKEN writing permission
```
Comment (antonioaversa) [inline .github/workflows/dogfood.yml:19]: Educational: why do we need the `contents` permission?
sonar-java doesn't seem to require it, and has a very similar yml, see [here](https://github.com/SonarSource/sonar-java/blob/master/.github/workflows/dogfood.yml).

**Review by antonioaversa — APPROVED** [review]
LGTM!
Minor, non-blocking remarks.

File: .github/workflows/dogfood.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+name: dogfood merge
+# This workflow is triggered on pushes to master and dogfood branches
+on:
```
Comment (antonioaversa) [inline .github/workflows/dogfood.yml:3]: Educational: I guess this `on` cannot be completely removed, without making the `yml` invalid, right?

File: .github/workflows/dogfood.yml (hunk 2 above)
Comment (Tim-Pohlmann) [reply .github/workflows/dogfood.yml:48]: I'll keep it in sync.

File: .github/workflows/dogfood.yml (hunk 3 above)
Comment (Tim-Pohlmann) [reply .github/workflows/dogfood.yml:22]: I have no idea. The python one uses it. I'll remove it and we see if it works.

File: .github/workflows/dogfood.yml (hunk 4 above)
Comment (Tim-Pohlmann) [reply .github/workflows/dogfood.yml:3]: Why would you want to remove it?

File: .github/workflows/dogfood.yml (hunk 1 above)
Comment (antonioaversa) [reply .github/workflows/dogfood.yml]: Evgeny has a more general, repo-independent way: `args: "Dogfood merge failed by {{ GITHUB_ACTOR }}, see {{ GITHUB_SERVER_URL }}/{{ GITHUB_REPOSITORY }}/actions/runs/{{ GITHUB_RUN_ID }}"`

File: .github/workflows/dogfood.yml (hunk 4 above)
Comment (antonioaversa) [reply .github/workflows/dogfood.yml:3]: Nevermind, you still need to tell on which actions. All good, I misunderstood the section.

**Pull request merged**